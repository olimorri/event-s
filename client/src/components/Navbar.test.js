import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NavBar from './Navbar';

const NavBarTest = () => (
  <BrowserRouter>
    <NavBar />
  </BrowserRouter>
);

describe('Snapshot', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<NavBarTest />);
    expect(asFragment(<NavBarTest />)).toMatchSnapshot();
  });
});

describe('Unit Testing', () => {
  it('should render the home button', () => {
    render(<NavBarTest />);
    const homeButton = screen.getByRole('button', { name: /home/i });
    expect(homeButton).toBeInTheDocument();
  });

  it('should render the events button', () => {
    render(<NavBarTest />);
    const homeButton = screen.getByRole('button', { name: /events/i });
    expect(homeButton).toBeInTheDocument();
  });
});
