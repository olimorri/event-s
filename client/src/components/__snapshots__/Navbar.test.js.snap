// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1l7wwbw"
  >
    <div
      class="css-1msacc8"
    >
      <a
        href="/"
      >
        <img
          alt=""
          class="chakra-image__placeholder css-gw1u79"
        />
      </a>
      <div
        class="css-17xejub"
      />
      <div
        class="chakra-stack css-det3q5"
      >
        <nav
          class="chakra-stack css-dss5yb"
        >
          <a
            href="/"
          >
            <button
              class="chakra-button css-1h2uyey"
              type="button"
            >
              Home
            </button>
          </a>
          <a
            href="/events"
          >
            <button
              class="chakra-button css-1h2uyey"
              type="button"
            >
              Events
            </button>
          </a>
        </nav>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="css-70qvj9"
      >
        <button
          aria-controls="menu-list-3"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-button chakra-menu__menu-button css-1nyg99m"
          id="menu-button-3"
          type="button"
        >
          <span
            class="css-a9hmjk"
          >
            <span
              class="chakra-avatar css-1ruhx4w"
            >
              <svg
                class="css-16ite8i"
                role="img"
                viewBox="0 0 128 128"
              >
                <path
                  d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                  fill="currentColor"
                />
                <path
                  d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                  fill="currentColor"
                />
              </svg>
            </span>
          </span>
        </button>
        <div
          class="css-0"
          style="position: absolute; left: 0px; top: 0px; visibility: hidden;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-1jj07fx"
            id="menu-list-3"
            role="menu"
            style="transform-origin: bottom right; opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <a
              href="/login"
            >
              <button
                class="chakra-menu__menuitem css-4dys5b"
                id="menuitem-1"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                Sign In
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
